lib/stripe/stripe_client.rb

```diff
@@ -14,28 +14,29 @@ class StripeClient
     @last_connection_manager_gc = Util.monotonic_time
 
     # Initializes a new StripeClient
-    def initialize(config_overrides = {})
+    def initialize(config_arg = {})
       @system_profiler = SystemProfiler.new
       @last_request_metrics = nil
 
-      # Supports accepting a connection manager object for backwards
-      # compatibility only, and that use is DEPRECATED.
-      @config_overrides = case config_overrides
-                          when Stripe::ConnectionManager
-                            {}
-                          when String
-                            { api_key: config_overrides }
-                          else
-                            config_overrides
-                          end
-    end
-
-    # Always base config off the global Stripe configuration to ensure the
-    # client picks up any changes to the config.
-    def config
-      Stripe.configuration.reverse_duplicate_merge(@config_overrides)
-    end
-
+      @config = case config_arg
```

Comment:
See comment above, but eliminated the "lazy" configuration initialization with always setting `@config` directly on initialization.